client/client.go

```diff
@@ -118,10 +118,11 @@ type ApiClient struct {
 }
 
 // Return a new ApiRequest sharing this ApiClient helper
-func (a *ApiClient) Request(code AuthToken) *ApiRequest {
+func (a *ApiClient) Request(code AuthToken, req func() (AuthToken, error)) *ApiRequest {
```

Comment:
Not a big issue, but we could have something like:
`type ClientReauthorizeFunc func() (AuthToken, error)
func (a *ApiClient) Request(code AuthToken, req ClientReauthorizeFunc) *ApiRequest `
This might be more readable as we are using this function signature in many places.